types/wordpress__blocks/index.d.ts

```diff
@@ -0,0 +1,403 @@
+// Type definitions for @wordpress/blocks 6.4
+// Project: https://github.com/WordPress/gutenberg/tree/master/packages/blocks/README.md
+// Definitions by: Derek Sifford <https://github.com/dsifford>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.5
+
+import { Dashicon } from '@wordpress/components';
+import { dispatch, select } from '@wordpress/data';
+import { ComponentType } from '@wordpress/element';
+
+export * from './api';
+export { withBlockContentContext } from './block-content-provider';
+
+declare module '@wordpress/data' {
+    function dispatch(key: 'core/blocks'): typeof import('./store/actions');
+    function select(key: 'core/blocks'): typeof import('./store/selectors');
+}
+
+export type BlockAlignment = 'left' | 'center' | 'right' | 'wide' | 'full';
+
+export interface BlockEditProps<T extends Record<string, any>> extends BlockSaveProps<T> {
+    readonly className: string;
```

Comment:
Apparently, we now also have access to the following properties sent to the edit function:
`clientId`,
`insertBlocksAfter`,
`isSelectionEnabled`,
`mergeBlocks`,
`name`,
`onReplace` and
`toggleSelection`.

Reply:
No time at the moment to get to this, but feel free to handle and open a PR.